sdk/compact_test.go

```diff
@@ -0,0 +1,680 @@
+package sdk_test
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"io"
+	"testing"
+	"time"
+
+	"github.com/convox/rack/pkg/options"
+	"github.com/convox/rack/pkg/structs"
+	"github.com/convox/rack/sdk"
+	"github.com/convox/stdapi"
+	"github.com/gorilla/websocket"
+	"github.com/stretchr/testify/require"
+)
+
+func TestAppParametersGet(t *testing.T) {
+	app := "app1"
+	params := map[string]string{
+		"hello":  "world",
+		"convox": "rocks",
+	}
+
+	s := stdapi.New("api", "api")
+	s.Route("GET", fmt.Sprintf("/apps/%s/parameters", app), func(c *stdapi.Context) error {
+		return c.RenderJSON(params)
+	})
+
+	testServer(t, s, func(c *sdk.Client) {
+		got, err := c.AppParametersGet(app)
+		require.NoError(t, err)
+		require.Equal(t, params, got)
+	})
+}
+
+func TestAppParametersSet(t *testing.T) {
+	app := "app1"
+	params := map[string]string{
+		"hello":  "world",
+		"convox": "rocks",
+	}
+
+	s := stdapi.New("api", "api")
+	s.Route("POST", fmt.Sprintf("/apps/%s/parameters", app), func(c *stdapi.Context) error {
+		for k, v := range params {
+			require.Equal(t, v, c.Form(k))
+		}
+		return c.RenderOK()
+	})
+
+	testServer(t, s, func(c *sdk.Client) {
+		err := c.AppParametersSet(app, params)
+		require.NoError(t, err)
+	})
+}
+
+func TestBuildCreateUpload(t *testing.T) {
+	app := "app1"
+	build := structs.Build{
+		Id:  "1",
+		App: app,
+	}
+
+	s := stdapi.New("api", "api")
+	s.Route("POST", fmt.Sprintf("/apps/%s/builds", app), func(c *stdapi.Context) error {
+		require.Equal(t, "convox", c.Form("description"))
+		require.Equal(t, "true", c.Form("cache"))
+
+		fp, _, err := c.Request().FormFile("source")
+		require.NoError(t, err)
+		b, err := io.ReadAll(fp)
+		require.NoError(t, err)
+		require.Equal(t, "test", string(b))
+
+		return c.RenderJSON(build)
+	})
+
+	testServer(t, s, func(c *sdk.Client) {
+		got, err := c.BuildCreateUpload(app, bytes.NewReader([]byte("test")), structs.BuildCreateOptions{
```

Comment:
Can you move `bytes.NewReader([]byte("test")` to a variable? e.g. `expectedSource`.